os-server/src/main/java/app/openschool/course/CourseController.java

```diff
@@ -1,34 +1,55 @@
 package app.openschool.course;
 
+import app.openschool.common.response.ResponseMessage;
 import app.openschool.course.api.dto.CourseDto;
 import app.openschool.course.api.dto.CourseInfoDto;
+import app.openschool.course.api.dto.CreateCourseRequest;
+import app.openschool.course.api.dto.UpdateCourseRequest;
 import app.openschool.course.api.mapper.CourseMapper;
+import app.openschool.user.User;
+import app.openschool.user.UserRepository;
 import io.swagger.v3.oas.annotations.Operation;
 import io.swagger.v3.oas.annotations.Parameter;
 import io.swagger.v3.oas.annotations.media.Content;
 import io.swagger.v3.oas.annotations.media.Schema;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
 import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import java.security.Principal;
 import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
-
 @RestController
 @RequestMapping("/api/v1/courses")
 public class CourseController {
 
   private final CourseService courseService;
+  private final UserRepository userRepository;
+  private final CourseRepository courseRepository;
```

Comment:
do not inject repositories directly in the controller, you have service layer for that